src/cargo/util/errors.rs

```diff
@@ -286,6 +286,37 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool {
+        match self.class() {
+            git2::ErrorClass::Net |
+            git2::ErrorClass::Os => true,
```

Comment:
This is the class I got for my test when it timed out.  When I tell the test to connect to a server that is not running https://github.com/rust-lang/cargo/pull/2396/files#diff-5a6267705b81a22950ae48ebdce1a553R8 

I thought it would have gave a ```Net``` error but ```Os``` is what it printed.